File: python/countdown-thread.py
import time
from threading import Thread

NUMBER = 10 ** 9  # 1 Billion


def countdown(n):
    while n > 0:
        n = n - 1


t0 = Thread(target=countdown, args=(NUMBER // 2,))
t1 = Thread(target=countdown, args=(NUMBER // 2,))

print("Counting down two threads with", NUMBER // 2)

start = time.time()
t0.start()
t1.start()
t0.join()
t1.join()
end = time.time()

print("Took {:f}s".format(end - start))

File: rust/countdown-thread.rs
use std::time::{Instant};
use std::thread;

static NUMBER: u32 = 1_000_000_000;

fn countdown(num: u32) {
    let mut i = num;
    while i > 0 {
        i -= 1;
    }
}

fn main() {
    println!("Counting down two threads with {:?}", NUMBER/2);
    let start_time = Instant::now();
    let thread1 = thread::spawn(move || {
        countdown(NUMBER/2);
    });
    let thread2 = thread::spawn(move || {
        countdown(NUMBER/2);
    });
    thread1.join().unwrap();
    thread2.join().unwrap();
    let elapsed_time = start_time.elapsed();
    println!("Took {:?}", elapsed_time);
}

